### DISCUSSION TITLE
Idea - allow to import private types (but not to use them)

### DISCUSSION THREAD

**Discussion by mac-hel** [vlang/v#20790] [Ideas]
I think it could be handy to be able to import struct (maybe enum) that is not public - just to access public static methods implemented on them
Ofcourse creation of private struct directly would be still forbidden

### Now
struct is private, can't import it. Can get to 'new' only via module name:
```v
import my_module
fn main() {
	input := my_module.Input.new('params', 'data')
}
```

### Proposal
Because 'new' is public (despite 'Input' is private) I would like to use it without going through module name:
```v
import my_module {Input}	          // now can't - Input is private
fn main() {
	input := Input.new('params', 'data')	        // would be fine, 'new' is public
	input2 := Input {'params','data'}		// should give ERROR - 'Input' is private
}
```

import should allow bringing private structs into scope
privacy rules should be enforced when imported struct is used

**Comment by Casper64** [comment]
Apart from whether it should be allow or not, I think it should not allow `import { Input }` but rather `import { Input.new }`. So only import the function not the whole struct.

Because a static method is basically just a function, being able to import the whole struct and only use the public (static) methods seems confusing.
